bindings/fcl-js.go

```diff
@@ -0,0 +1,133 @@
+package bindings
+
+import (
+	"bytes"
+	"os"
+	"path/filepath"
+	"runtime"
+	"sort"
+	"text/template"
+
+	"github.com/onflow/flixkit-go"
+	"github.com/stoewer/go-strcase"
+)
+
+type simpleParameter struct {
+	Name        string
+	JsType      string
+	Description string
+	FclType     string
+	CadType     string
+}
+
+type templateData struct {
+	Version         string
+	Parameters      []simpleParameter
+	Title           string
+	Description     string
+	Location        string
+	IsScript        bool
+	IsLocalTemplate bool
+}
+
+type FclJSGenerator struct {
+	TemplateDir string
+}
+
+func NewFclJSGenerator() *FclJSGenerator {
+	_, currentFilePath, _, _ := runtime.Caller(0)
+	baseDir := filepath.Dir(currentFilePath)
+	templateDir := filepath.Join(baseDir, "templates")
+
+	return &FclJSGenerator{
+		TemplateDir: templateDir,
+		// initialize other fields if needed
+	}
+}
+
+func (g FclJSGenerator) Generate(flix *flixkit.FlowInteractionTemplate, templateLocation string, isLocal bool) (string, error) {
+	files, err := os.ReadDir(g.TemplateDir)
+	if err != nil {
+		return "", err
+	}
+	templateFiles := make([]string, 0)
+	for _, file := range files {
+		if !file.IsDir() {
+			templateFiles = append(templateFiles, g.TemplateDir+"/"+file.Name())
```

Comment:
Use `filepath.Join` here or you might get issues on different platforms

Reply:
filepath.Walk gives you files with path. https://golang.cafe/blog/how-to-list-files-in-a-directory-in-go.html

Reply:
Thanks, updated PR to use filepath.Walk